source/shared-union-fields-deep.d.ts

```diff
@@ -83,7 +83,15 @@ function displayPetInfo(petInfo: SharedUnionFieldsDeep<Cat | Dog>['info']) {
 export type SharedUnionFieldsDeep<Union, Options extends SharedUnionFieldsDeepOptions = {recurseIntoArrays: false}> =
 // If `Union` is not a union type, return `Union` directly.
```

Comment:
I kind of forgot why I limited the input to `union`...

If I understand your question correctly, just need to remove that limit 
```typescript
IsUnion<Union> extends false ? Union
```

Reply:
You're right, removing this conditional gives the same results. I updated my PR with this simplification 👍